cas_client/src/interface.rs

```diff
@@ -57,14 +57,7 @@ pub trait Client {
     ) -> Result<Option<Bytes>>;
 
     /// Upload a new shard.
-    async fn upload_shard(
-        &self,
-        prefix: &str,
-        hash: &MerkleHash,
-        force_sync: bool,
-        shard_data: bytes::Bytes,
-        salt: &[u8; 32],
-    ) -> Result<bool>;
+    async fn upload_shard(&self, shard_data: Bytes, force_sync: bool, salt: &[u8; 32]) -> Result<bool>;
```

Comment:
Looking at the code, we might be able to remove the `force_sync` and `salt` params from this function as they aren't really used (I'm fine with removing salt as a separate PR since it might require a bit more refactoring for the local_client).